lib/uklock/mutex.c

```diff
@@ -1,8 +1,39 @@
 #include <uk/mutex.h>
 
+#ifdef CONFIG_LIBUKLOCK_MUTEX_METRICS
+#include <uk/init.h>
+
+struct uk_mutex_metrics _uk_mutex_metrics;
+__spinlock              _uk_mutex_metrics_lock;
+#endif /* CONFIG_LIBUKLOCK_MUTEX_METRICS */
+
 void uk_mutex_init(struct uk_mutex *m)
 {
 	m->lock_count = 0;
 	m->owner = NULL;
 	uk_waitq_init(&m->wait);
+
+#ifdef CONFIG_LIBUKLOCK_MUTEX_METRICS
+	ukarch_spin_lock(&_uk_mutex_metrics_lock);
+	_uk_mutex_metrics.active_unlocked++;
+	ukarch_spin_unlock(&_uk_mutex_metrics_lock);
+#endif /* CONFIG_LIBUKLOCK_MUTEX_METRICS */
+}
+
```

Comment:
```suggestion
void uk_mutex_get_metrics(struct uk_mutex_metrics *dst)
{
	UK_ASSERT(dst);

	ukarch_spin_lock(&_uk_mutex_metrics_lock);
	memcpy(dst, &_uk_mutex_metrics, sizeof(*dst));
	ukarch_spin_unlock(&_uk_mutex_metrics_lock);
}
```

A separate function can do the retrieval so users do not use the `_uk_mutex_metrics` directly.

Reply:
You might need to include something `#include <uk/assert.h>`